### DISCUSSION TITLE
EnableIdempotence is not working

### DISCUSSION THREAD

**Discussion by Marmosquera** [Farfetch/kafkaflow#294] [Q&A]
Hi! I'm creating the producer with:

```
            var _idempotentProducerConfig = new Confluent.Kafka.ProducerConfig
            {
                EnableIdempotence = true
            };

            return builder
                    .WithProducerConfig(_idempotentProducerConfig)
                    .WithAcks(KafkaFlow.Acks.All)
                    .DefaultTopic(kafkaTopic)
                    .AddMiddlewares(middlewares => middlewares
                        .Add<TracingMiddleware>()
                        .Add<LoggingMiddleware>()
                        .AddSerializer<JsonCoreSerializer>()
                        );
```

And I'm sending the message with:
```
        private async Task ProduceAsync<TMess>(TMess message, Guid id)
        {
            await _producers[PRODUCER_NAME].ProduceAsync(id.ToString(), message);
        }
```

But I'm getting the messages duplicated (same id) on the topic. What I'm missing?

**Comment by filipeesch** [comment]
Hi @Marmosquera ,

I think this problem is related to how the Confluent Driver works when idempotence is enabled. When producing Kafka Flow just execute the middleware and call de driver's produce methods. See he driver settings: https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html#producerconfigs_enable.idempotence

**Comment by Marmosquera** [comment]
Thx @filipeesch for your answer ... but I asked to confluent-kafka-dotnet team and they said that I should ask kafka-flow.
I'll keep trying to make it work.
